### PR TITLE
fix #222 Browsers block data URI

### LINKED ISSUE #222: Browsers block data URI
When clicking "Save Image" SVG-edit will navigate to a new tab opening a data URI. Firefox 58 will block this and as I understand other browsers will do the same.

### PR DESCRIPTION
**Pull request by cuixiping**
data url is unsafe and blocked by firefox and chrome, so use blob url instead, old browsers will fallback to data url.

### CODE REVIEW

File: editor/svgutils.js [hunk 1]
```diff
@@ -170,6 +170,45 @@ svgedit.utilities.encodeUTF8 = function (argString) {
...
+ * @param {string} dataurl
+ * @return {string} object URL or empty string
+ */
+svgedit.utilities.dataURLToObjectURL = function (dataurl) {
+	if (typeof Uint8Array == 'undefined' || typeof Blob == 'undefined' || typeof URL == 'undefined' || !URL.createObjectURL) {
+		return '';
+	}
+    var arr = dataurl.split(','), mime = arr[0].match(/:(.*?);/)[1],
+        bstr = atob(arr[1]), n = bstr.length, u8arr = new Uint8Array(n);
+    while(n--){
```
Comment (codedread) [inline editor/svgutils.js]: nit: Please add spaces:

  while (n--) {

File: editor/svgutils.js [hunk 2]
```diff
@@ -170,6 +170,45 @@ svgedit.utilities.encodeUTF8 = function (argString) {
...
+        u8arr[n] = bstr.charCodeAt(n);
+    }
+    var blob = new Blob([u8arr], {type:mime});
+	return URL.createObjectURL(blob);
+};
```
Comment (codedread) [inline editor/svgutils.js]: nit: add a blank line between methods

**Comment by cuixiping** [comment]
I added spaces and blank lines at fix-#222-browsers-block-dataURI branch on cuixiping/svgedit.
Should I send pull request again?

**Comment by codedread** [comment]
Thanks for the PR!

**Pull request merged**